sdk/core/azure-core/inc/azure/core/http/raw_response.hpp

```diff
@@ -76,10 +76,19 @@ namespace Azure { namespace Core { namespace Http {
             response.m_statusCode,
             response.m_reasonPhrase)
     {
+      AZURE_ASSERT(m_bodyStream == nullptr);
```

Comment:
"This type is copyable except that sometimes copying them explodes and kills the program" doesn't feel like a great design to me. Could we use the type system to make raw responses that can be copied distinct from those that can't?

Reply:
A comment that this peculiar design is necessary to workaround #2281 would be a good idea. I hadn't noticed the issue before performing my review, I suspect other underinformed folks in the future will also wonder what's going on here.